nmpolicy/internal/resolver/filter.go

```diff
@@ -40,29 +41,15 @@ func filter(inputState map[string]interface{}, path ast.VariadicOperator, expect
 	return filteredMap, nil
 }
 
-func isEqual(obtainedValue interface{}, desiredValue ast.Node) (bool, error) {
-	if desiredValue.String != nil {
-		stringToCompare, ok := obtainedValue.(string)
-		if !ok {
-			return false, fmt.Errorf("the value %v of type %T not supported,"+
-				"curretly only string values are supported", obtainedValue, obtainedValue)
-		}
-		return stringToCompare == *desiredValue.String, nil
-	}
-
-	return false, fmt.Errorf("the desired value %v is not supported. Curretly only string values are supported", desiredValue)
-}
-
-func mapContainsValue(mapToFilter map[string]interface{}, filterKey string, expectedNode ast.Node) (interface{}, error) {
+func mapContainsValue(mapToFilter map[string]interface{}, filterKey string, expectedValue interface{}) (interface{}, error) {
 	obtainedValue, ok := mapToFilter[filterKey]
 	if !ok {
 		return nil, fmt.Errorf("cannot find key %s in %v", filterKey, mapToFilter)
 	}
-	valueIsEqual, err := isEqual(obtainedValue, expectedNode)
-	if err != nil {
-		return nil, fmt.Errorf("error comparing the expected and obtained values : %v", err)
+	if reflect.TypeOf(obtainedValue) != reflect.TypeOf(expectedValue) {
+		return nil, fmt.Errorf(`type missmatch: "%T" != "%T" -> %+v != %+v`, obtainedValue, expectedValue, obtainedValue, expectedValue)
```

Comment:
Please make the error message more informative to the user.
`type missmatch: "The value in the path doesn't match the value to filter. %T" != "%T" -> %+v != %+v`

Reply:
Done